app/Models/Foundation/Summit/Events/SummitEvent.php

```diff
@@ -1739,6 +1764,160 @@ public function promote2Presentation(PresentationType $type):void{
         } catch (\Exception $ex) {
 
         }
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getOverflowStreamingUrl(): ?string
+    {
+        return $this->overflow_streaming_url;
+    }
+
+    /**
+     * @return bool
+     */
+    public function gatOverflowStreamIsSecure(): bool
+    {
+        return $this->overflow_stream_is_secure;
+    }
+
+    /**
+     * @return string|null
+     */
+    public function getOverflowStreamKey(): ?string
+    {
+        return $this->overflow_stream_key;
+    }
+
+    /**
+     * @param string $overflow_stream_key
+     * @return void
+     */
+    public function setOverflowStreamKey(string $overflow_stream_key): void
+    {
+        $this->overflow_stream_key = $overflow_stream_key;
+    }
+
+    /**
+     * @param string $overflow_streaming_url
+     * @param bool $overflow_stream_is_secure
+     * @return void
+     */
+    public function setOverflow(string $overflow_streaming_url,
+                                bool $overflow_stream_is_secure): void
+    {
+        $this->overflow_streaming_url = $overflow_streaming_url;
+        $this->overflow_stream_is_secure = $overflow_stream_is_secure;
+        $this->occupancy = self::OccupancyOverflow;
+    }
+
+    public function clearOverflow(string $occupancy = self::OccupancyEmpty): void
+    {
+        $this->overflow_streaming_url = null;
+        $this->overflow_stream_is_secure = false;
+        $this->overflow_stream_key = null;
+        $this->occupancy = $occupancy;
+    }
+
+    /**
+     * @return string
+     * @throws ValidationException
+     */
+    public function getOverflowUrl(): string
+    {
+        if ($this->occupancy != self::OccupancyOverflow)
+            throw new ValidationException("To get the overflow url, occupancy must be OVERFLOW.");
+
+        if (is_null($this->overflow_stream_key))
+            throw new ValidationException("Overflow stream key is null.");
+
+        return sprintf("%s%s?%s=%s",
+            $this->summit->getMarketingSiteUrl(),
+            config("overflow.path", "/a/overflow-player"),
+            config("overflow.query_string_key","k"),
+            $this->overflow_stream_key);
+    }
+
+    /**
+     * @return string
+     * @throws RandomException
+     */
+    public function generateOverflowKey(): string
+    {
+         $salt = random_bytes(16);
+         return hash('sha256', $this->getId() . $salt . time());
+    }
+
+    /**
+     * @param bool $overflow
+     * @return array
+     */
+    public function getStreamingTokens(bool $overflow = true): array
+    {
+        $tokens = [];
+        $summit = $this->summit;
+        $cache_key = $this->getSecureStreamCacheKey();
```

Comment:
the cache key is not the same 
u need to have one for regular stream and another one for overflow stream